### PR TITLE
opentelemetry plugin

### PR DESCRIPTION
**Pull request by brig**
plugin configuration is set as the default task variables for the `opentelemetry` task
```
opentelemetry:
  enabled: true
  endpoint: "http://tempo-distributor.local:4318/v1/traces"   # OTLP traces endpoint
  link: "https://monitoring.example.com/explore?left=%7B%22range%22%3A%7B%22from%22%3A%22now-7d%22%2C%22to%22%3A%22now%22%7D%2C%22datasource%22%3A%22tempo_traces%22%2C%22queries%22%3A%5B%7B%22query%22%3A%22<traceId>%22%2C%22queryType%22%3A%22traceql%22%7D%5D%7D" # optional link template where to find traces...
```
Screenshot 2024-10-16 at 11 18 03

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -52,6 +52,7 @@
         <module>tasks/terraform</module>
         <module>tasks/xml</module>
         <module>tasks/zoom</module>
+        <module>tasks/opentelemetry</module>
```
[the remark this answers is not in the document]
Comment (brig) [reply pom.xml:55]: oh, yeah, why not :)

File: tasks/opentelemetry/pom.xml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            <artifactId>jackson-annotations</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.dataformat</groupId>
+            <artifactId>jackson-dataformat-yaml</artifactId>
+        </dependency>
```
Comment (benbroadaway) [inline tasks/opentelemetry/pom.xml:90]: jackson deps can be `provided` scope

Comment (brig) [reply tasks/opentelemetry/pom.xml:90]: We should mention somewhere in the documentation what dependencies and how can be used in plugins :)

Comment (benbroadaway) [reply tasks/opentelemetry/pom.xml]: We [kind of do mention it](https://concord.walmart.com/docs/processes-v2/tasks.html#:~:text=Some%20dependencies%20are,slf4j/slf4j%2Dapi) but I agree we need to give a more explicit here-exactly-what's-`provided` <thing>.

Comment (brig) [reply tasks/opentelemetry/pom.xml]: Oh, right, I didn't notice :)

File: runtime/opentelemetry/src/main/java/com/walmartlabs/concord/plugins/opentelemetry/OpentelemetryModule.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+import com.google.inject.Binder;
+import com.google.inject.multibindings.Multibinder;
+import com.walmartlabs.concord.svm.ExecutionListener;
+
+import javax.inject.Named;
+
+@Named
+public class OpentelemetryModule implements com.google.inject.Module {
```
Comment (ibodrov) [inline runtime/opentelemetry/src/main/java/com/walmartlabs/concord/plugins/opentelemetry/OpentelemetryModule.java:30]:
```suggestion
public class OpentelemetryModule implements Module {
```

**Comment by ibodrov** [comment]
Let's add a README.md with the usage instructions. We can link to https://www.jaegertracing.io/docs/1.62/getting-started/#all-in-one or w/e for a local collector.

File: tasks/opentelemetry/pom.xml (hunk 2 above)
Comment (brig) [reply tasks/opentelemetry/pom.xml]: we need maven plugin for concord plugins :)
so maven plugin can verify that everything is ok with plugin...

**Review by ibodrov — APPROVED** [review]

**Pull request merged**